lib/json5/lexer.ml

```diff
@@ -0,0 +1,202 @@
+  | CLOSE_BRACKET -> Format.fprintf ppf "']'"
+  | COLON -> Format.fprintf ppf "':'"
+  | COMMA -> Format.fprintf ppf "','"
+  | COMMENT s -> Format.fprintf ppf "COMMENT '%s'" s
+  | TRUE -> Format.fprintf ppf "'true'"
+  | FALSE -> Format.fprintf ppf "'false'"
+  | NULL -> Format.fprintf ppf "'null'"
+  | FLOAT s -> Format.fprintf ppf "FLAOT '%s'" s
+  | INT_OR_FLOAT s -> Format.fprintf ppf "INT_OR_FLOAT '%s'" s
+  | INT s -> Format.fprintf ppf "INT '%s'" s
+  | STRING s -> Format.fprintf ppf "STRING '%s'" s
+  | IDENTIFIER_NAME s -> Format.fprintf ppf "IDENTIFIER_NAME '%s'" s
+
+let source_character = [%sedlex.regexp? any]
+let line_terminator = [%sedlex.regexp? 0x000A | 0x000D | 0x2028 | 0x2029]
+
+let line_terminator_sequence =
+  [%sedlex.regexp? 0x000A | 0x000D, Opt 0x000A | 0x2028 | 0x2029]
+
+(* NUMBERS, 7.8.3 *)
+let non_zero_digit = [%sedlex.regexp? '1' .. '9']
+let decimal_digit = [%sedlex.regexp? '0' .. '9']
+let decimal_digits = [%sedlex.regexp? Plus decimal_digit]
+let hex_digit = [%sedlex.regexp? '0' .. '9' | 'a' .. 'f' | 'A' .. 'F']
+let exponent_indicator = [%sedlex.regexp? 'e' | 'E']
+
+let signed_integer =
+  [%sedlex.regexp? decimal_digits | '+', decimal_digits | '-', decimal_digits]
+
+let exponent_part = [%sedlex.regexp? exponent_indicator, signed_integer]
+
+let decimal_integer_literal =
+  [%sedlex.regexp? '0' | non_zero_digit, Opt decimal_digits]
+
+let hex_integer_literal =
+  [%sedlex.regexp? "0x", Plus hex_digit | "0X", Plus hex_digit]
+
+(* float *)
+let float_literal =
+  [%sedlex.regexp?
+    ( decimal_integer_literal, '.', Opt decimal_digits, Opt exponent_part
+    | '.', decimal_digits, Opt exponent_part )]
+
+let json5_float =
+  [%sedlex.regexp? float_literal | '+', float_literal | '-', float_literal]
+
+(* int_or_float *)
+let int_or_float_literal =
+  [%sedlex.regexp? decimal_integer_literal, Opt exponent_part]
+
+let json5_int_or_float =
+  [%sedlex.regexp?
+    int_or_float_literal | '+', int_or_float_literal | '-', int_or_float_literal]
+
+(* int/hex *)
+let int_literal =
+  [%sedlex.regexp? decimal_digits | '+', decimal_digits | '-', decimal_digits]
+
+let json5_int =
+  [%sedlex.regexp?
+    ( hex_integer_literal
+    | '+', hex_integer_literal
+    | '-', hex_integer_literal
+    | int_literal )]
+
+(* STRING LITERALS, 7.8.4 *)
+let unicode_escape_sequence =
+  [%sedlex.regexp? 'u', hex_digit, hex_digit, hex_digit, hex_digit]
+
+let single_escape_character = [%sedlex.regexp? Chars {|'"\\bfnrtv|}]
+
+let escape_character =
+  [%sedlex.regexp? single_escape_character | decimal_digit | 'x' | 'u']
+
+let non_escape_character =
+  [%sedlex.regexp? Sub (source_character, (escape_character | line_terminator))]
+
+let character_escape_sequence =
+  [%sedlex.regexp? single_escape_character | non_escape_character]
+
+let line_continuation = [%sedlex.regexp? '\\', line_terminator_sequence]
+
+let escape_sequence =
+  [%sedlex.regexp? character_escape_sequence | '0' | unicode_escape_sequence]
+(* TODO *)
```

Comment:
There is still this TODO.

Reply:
I remove the TODO it seems to be remaining.